be/src/exec/pipeline/scan/connector_scan_operator.cpp

```diff
@@ -174,6 +415,7 @@ ConnectorChunkSource::ConnectorChunkSource(ScanOperator* op, RuntimeProfile* run
     _data_source->set_read_limit(_limit);
     _data_source->set_runtime_profile(runtime_profile);
     _data_source->update_has_any_predicate();
+    _op = down_cast<ConnectorScanOperator*>(op);
```

Comment:
why does chunk source depend on scan operator

Reply:
because chunk source has to pass some runtime information back to scan operator, and let scan operator to decide how many io tasks to issue in next period.

Reply:
can we expose those information by providing a API of chunk source